integration-tests/test/node_client.py

```diff
@@ -0,0 +1,196 @@
+import base64
+import logging
+import socket
+from concurrent import futures
+from contextlib import contextmanager
+from queue import Queue
+from typing import Generator, Iterator
+import os
+import subprocess
+from pathlib import Path
+import grpc
+from cryptography.hazmat.backends import default_backend
+from cryptography.hazmat.backends.openssl.ec import _EllipticCurvePrivateKey
+from cryptography.hazmat.primitives.serialization import load_pem_private_key
+from docker import DockerClient
+from eth_hash.auto import keccak
+from rchain.pb import CasperMessage_pb2
+from rchain.pb.CasperMessage_pb2 import BlockMessage, BlockRequest
+from rchain.pb.routing_pb2 import (Chunk, ChunkResponse, Header, Node, NoResponse,
+                             Packet, Protocol, TLRequest, TLResponse)
+from rchain.pb.routing_pb2_grpc import (TransportLayerServicer, TransportLayerStub,
+                                  add_TransportLayerServicer_to_server)
+
+from .rnode import Node as RNode
+
+DEFAULT_TRANSPORT_SERVER_PORT = 40400
+DEFAULT_NETWORK_ID = 'testnet'
+
+logger = logging.getLogger("node_client")
+
+
+def get_node_id_raw(key: _EllipticCurvePrivateKey) -> bytes:
+    W = key.public_key().public_numbers()
+    pk_bytes = W.x.to_bytes(32, 'big') + W.y.to_bytes(32, 'big')
+    return keccak(pk_bytes)[12:]
```

Comment:
Perhaps this function should be part of `pyrchain`?  What do you think?

Reply:
Yeah. That would be better. But currently, there is no such a function.

Reply:
I think we can leave it here now.

Reply:
>  But currently, there is no such a function.
What's stopping you from adding it there?

Reply:
If you want to push this PR out ASAP, let's ticket the move.

Reply:
@adaszko Are you going to create the ticket? AFAIK this was taken from https://github.com/rchain/rchain-testnet-node/tree/dev/scripts and AFAIK there are currently no plans nor requirements for pyrchain.

Reply:
Are there any requirements for pyrchain at all?  I see pyrchain as a Python rnode driver.  That includes any code that's necessary to use it, like `get_node_id_raw()` seems to be.

Reply:
There probably aren't any requirements at all. It emerged organically as I was abstracting common things needed for wallet transfer test. Again, it'd be nice if it evolved into full blown RNode compatible library, but AFAIK officially no such work is planned. That doesn't stop anyone from doing it and reaping the using the results elsewhere though.

Reply:
What's planned isn't really relevant here.  What's important is that code like `get_node_id_raw()` doesn't belong to integration tests because it makes assumptions about implementation details (note the need for additional dependencies that weren't needed so far: ecdsa, cryptography, eth-hash, pycryptodome) whereas tests should only make assumptions about the (CLI/Docker/Shell) interface.  Assumptions about implementation incur maintenance burden and should be delegated to some other place. `pyrchain` seems like an ideal candidate.

Reply:
True. You're right.